source/extensions/common/wasm/wasm_vm.cc

```diff
@@ -72,11 +72,27 @@ bool EnvoyWasmVmIntegration::getNullVmFunction(std::string_view function_name, b
   return false;
 }
 
+absl::string_view getWasmEngine() {
+  absl::string_view wasm_ext_runtime;
+  for (const auto& ext : Envoy::Registry::FactoryCategoryRegistry::registeredFactories()) {
+    if (ext.first == "envoy.wasm.runtime") {
+      for (auto engine_name : ext.second->registeredNames()) {
+        if (engine_name != "envoy.wasm.runtime.null") {
+          wasm_ext_runtime = engine_name;
+        }
```

Comment:
To future-proof this code, you might want to add `break` after finding the first non-null Wasm engine.

Reply:
Done.

Reply:
Actually, you should probably `return engine_name;` here, and `return "";` at the end of the function, otherwise you don't cover builds with Wasm disabled.